src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/pipeline.js

```diff
@@ -0,0 +1,145 @@
+import _ from 'lodash';
+import * as ProcessorTypes from './processor_types';
+
+export default class Pipeline {
+
+  constructor() {
+    this.processors = [];
+    this.counter = 0;
+    this.input = {};
+    this.output = undefined;
+    this.dirty = false;
+
+    this._processorTypes = _.reduce(ProcessorTypes, (result, Type, key) => {
+      const instance = new Type();
+      result[instance.typeId] = Type;
+      return result;
+    }, {});
+  }
+
+  get model() {
+    return {
+      input: this.input,
+      processors: _.map(this.processors, processor => processor.model)
+    };
+  }
+
+  load(pipeline) {
+    while (this.processors.length > 0) {
+      this.processors.pop();
```

Comment:
Still not sure why these need to be popped off individually instead of just creating a new empty array `this.processors = []`